core/src/main/java/io/questdb/cairo/TableReaderRecord.java

```diff
@@ -263,6 +260,22 @@ public long getUpdateRowId() {
         return getRowId();
     }
 
+    @Override
+    public long getUuidHi(int col) {
```

Comment:
one part at a time API is inefficient. This is the reason Long256 has a wrapper. We can do this via memory, e.g.  getUuidAddr(int col), then two Unsafe getLong() ?

Reply:
I [explored the idea](https://github.com/questdb/questdb/commit/9417647787fa9ee3852446b2d5a78892a02062d9) callers getting and re-using the address. 
What I did:
1. Added this method to `Record`:
```java
/**
 * Returns a location where the value of a given UUID column is stored.
 * The location is an opaque value and callers should not access it directly,
 * because it does not have to directly map to a memory address.
 * It is used to pass the location to {@link #getUuidHi(int, long)} and {@link #getUuidLo(int, long)}.
 *
 * @param col numeric index of the column
 * @return location of the UUID value
 */
default long getUuidLocation(int col) {
    throw new UnsupportedOperationException();
}
```
and then changed the Record `getHi()` and `getLo()` contract to:
```java
long getUuidHi(int col, long location);
long getUuidLo(int col, long location);
```
Question: Why `getUuidHi()` and `getUuidLo()` needs both `col` and `location`? Wouldn't be just the `location` enough? 
Answer: See for example the `JoinRecord`: The `col` is needed to select the right upstream Record.  

Further Findings: This new interface has to bubble down to Functions too. Why? Because UUID columns are often accessed via the `UuidColumn` function. This means to get the benefits from the address re-use `UuidColumn` users must get the address in their hands too and this forces us to expose the `getAddress()` in the Function interface. However, most functions do not need the address. For example `UuidConstant` is also a Function. So it has to implement the `getAddress()` function too. So it the experiment it returns a dummy constant. Which is then ignored. See: https://github.com/questdb/questdb/blob/9417647787fa9ee3852446b2d5a78892a02062d9/core/src/main/java/io/questdb/griffin/engine/functions/constants/UuidConstant.java#L32

So UUID Function users must first get the address and the pass it to `getHi()` / `getLo()` methods. Even when it's not needed at all. This does not look very nice and it also has some overhead. 

Moreover, it does not solve all cases of unnecessary work. For example this is how the `SelectedRecord` looks like:
```java
@Override
public long getUuidHi(int col, long location) {
    return base.getUuidHi(getColumnIndex(col), location);
}

@Override
public long getUuidLo(int col, long location) {
    return base.getUuidLo(getColumnIndex(col), location);
}

@Override
public long getUuidLocation(int col) {
    return base.getUuidLocation(getColumnIndex(col));
}
```
There are still 3 calls of `getColumnIndex()` to get a single UUID! `getUuidHi()` and `getUuidLo()` cannot access the location directly, because it is not necessary a memory address. For example when the `SelectedRecord` is backed by the `VirtualRecord` (=the record backed by functions) then chances are the `location` has no meaning at all!

The alternative is to introduce a wrapper, similar to `Long256`. Something like this:
```java
public UUIDWrapper getUuid(int col);
```
This way we would be spared of this misery. But that wrapper is not for free either. It means every function creating a UUID has to have an instance (or most likely 2 instances) of the wrapper, It also brings issues with function owning the wrapper and thread safety, etc. These disadvantages is currently not very visible in the code, because long256 has a limited support, but it's ther. We could also use this style of API:
```
public void getUuid(int col, UUIDSink); 
```
This way functions would not need to own the wrapper instances. But this would just move the ownership from functions to callers, so not really that different. 

The plan: I will benchmark the current implementation to see how it performs compared to the original code. And I would also appreciate your feedback. Also: Sorry for a very long message! :)